easybuild/framework/easyconfig/types.py

```diff
@@ -236,17 +247,47 @@ def to_dependency(dep):
         if not found_name_version:
             raise EasyBuildError("Can not parse dependency without name and version: %s", dep)
 
+    # also deal with dependencies in the "old" format
+    elif isinstance(dep, tuple):
+        depspec = dep
```

Comment:
this is a not a dict value (cfr. docstring)?

maybe use this?

```python
if isinstance(dep, (tuple, list)):
    if len(dep) >= 2:
        depspec.update({
            'name': dep[0],
            'version': dep[0],
        })
    if len(dep) >= 3:
        depspec.update({'versionsuffix': dep[2]})
    if len(dep) >= 4:
        depspec.update({'toolchain': dep[3]})
```

The rest of this function (see above) should handle the rest?